### ISSUE TITLE
AMDAVX architecture should cause USE_ISA_X86_64 to get set?

### ISSUE THREAD

**Issue by jjwilke** [kokkos/kokkos#2121]
I could totally be wrong - but is AMDAVX an x86-64 architecture and should it cause the ISA_X86_64 define to appear in the config header?

**Comment by nmhamster** [comment]
Yes it is an x86-64 architecture and yes it should cause ISA_X86_64 to appear. It reflects the issue that AMD's implementation of AVX may not be completely compatible with Intel generated code (since AVX on Intel sometimes means AVX + some other stuff).

**Comment by nmhamster** [comment]
@jjwilke - I think we should get rid of AMDAVX (looking at the code) as we want to switch to Ryzen, EPYC etc.

**Comment by dalg24** [comment]
> @jjwilke - I think we should get rid of AMDAVX (looking at the code) as we want to switch to Ryzen, EPYC etc.

What is the status of this?
Related to #2120

**Comment by jjwilke** [comment]
This was never resolved. We have *consistent* logic between `Makfile.kokkos` and `kokkos_arch.cmake`.  In `Makefile.kokkos`, we set `KOKKOS_INTERNAL_USE_ARCH_AVX` to true... which causes SandyBridge flags to get added.  I'm not sure what AMD processors would want us to set `Kokkos_ARCH_AMDAVX=ON` and if so if we should set sandybridge flags.  Someone more versed in AMD should check this.

**Comment by masterleinad** [comment]
It seems that we already set SandyBridge flags:
https://github.com/kokkos/kokkos/blob/3b83dec4edc3ae0a6a27dcd82a68513df619ad9c/cmake/kokkos_arch.cmake#L250-L258
Let's revisit this again after the release when we decide to treat Ryzen, EPYC etc. special.

**Comment by dalg24** [comment]
We will consider removing AMDAVX

**Comment by PhilMiller** [comment]
@dalg24 Should we revise this issue to just be 'Remove AMDAVX'?

**Comment by masterleinad** [comment]
> @dalg24 Should we revise this issue to just be 'Remove AMDAVX'?

Also, note that `USE_ISA_X86_64` was removed in 2461bcbfa8d94b77abdf57e3e8514aa2cdcb9b69.
